Kitodo/src/main/java/org/kitodo/production/services/data/FilterService.java

```diff
@@ -1047,4 +1047,38 @@ private Integer getTaskEnd(String parameter) {
         String[] strArray = parameter.split("-");
         return Integer.parseInt(strArray[1]);
     }
+
+    /**
+     * Parses the passed Map and returns the value. The Map should contain only one or no Entry.
+     * This method should be used to parse a Map passed by PrimeFaces' DataTable default filter functionality.
+     * It passes currently only one Entry because our custom filter input passes the complete filter string to
+     * the first input of PrimeFaces' default filter functionality.
+     * This triggers PrimeFaces to pass the filter string and update the DataTable automatically.
+     *
+     * @param filters
+     *      a Map containing only one String value
+     * @return
+     *      the only Entry's value as java.lang.String
+     * @throws IllegalArgumentException
+     *      when the passed Map contains more than one Entry or when the Entry does not contain a value of type String
+     *
+     */
+    String parsePrimeFacesFilter(Map filters) throws IllegalArgumentException {
+        if (Objects.nonNull(filters) && filters.entrySet().size() > 0) {
```

Comment:
Maybe this method could be simplified:

```
    /**
     * Parses the passed Map and returns the value. The Map should contain only one or no Entry.
     * This method should be used to parse a Map passed by PrimeFaces' DataTable default filter functionality.
     * It passes currently only one Entry because our custom filter input passes the complete filter string to
     * the first input of PrimeFaces' default filter functionality.
     * This triggers PrimeFaces to pass the filter string and update the DataTable automatically.
     *
     * @param filters
     *      a Map containing only one String value
     * @return
     *      the only Entry's value as java.lang.String
     */
    String parsePrimeFacesFilter(Map filters) {
        if (Objects.nonNull(filters) && filters.entrySet().size() == 1) {
            Object filter = filters.entrySet().stream().findAny();
            if (filter instanceof String) {
                return (String) filter;
            }
        }
        return "";
    }
```

Like I see in cases where this method is used, exception is caught and next filter is set up empty String.

This change could decrease amount of code needed later. Not needed catch of exceptions and check if filter is null.